nevergrad/instrumentation/variables.py

```diff
@@ -186,13 +187,13 @@ def data_to_argument(self, data: ArrayLike, deterministic: bool = False) -> Y:
         array = np.array(data, copy=False)
         for transf in self.transforms:
             array = transf.forward(array)
-        if self._asfloat:
-            return float(array[0])
+        if self._scalar_type is not None:
+            return self._scalar_type(array[0] if self._scalar_type != int else round(array[0]))
```

Comment:
Wouldn't it better to *cast* it to an int `int(array[0])` as python and numpy do instead of rounding?

e.g. for numpy
```
> np.array([1.9]).astype(int)
array([1])
```

Reply:
I would expect `Array(1).affined(a=1, b=12).asscalar(int)` to be sampled with N(12, 1) and therefore have mean 12, but if we cast to int, it would have mean 11.5 or something like this. Either way, there is a problem I think, and rounding seems the more legitimate... what do you think?

Reply:
Hum that's a very good point, your call on this one, you have more insight

Reply:
I'll go for rounding then, since it does not give any special importance to 0 compared to other values while casting to int does